APIImpl/Network.lua

```diff
@@ -0,0 +1,80 @@
+-- Network.lua
+
+-- Provides an implementation for the network API. Unlike Cuberites API this is blocking.
+
+
+
+
+
+local socket = require 'socket'
+local http = require 'socket.http'
+local https = require 'ssl.https'
+local ltn12 = require("ltn12")
+
+
+
+
+
+local function MakeRequest(a_Method, a_Url)
+	local respbody = {}
+	local protocol = a_Url:match("^https") and https or http
+	local success, status, headers = protocol.request({
+		url = a_Url,
+		method = a_Method,
+		sink = ltn12.sink.table(respbody),
+	})
+	respbody = table.concat(respbody)
+	success = (success ~= nil)
+	if (success) then
+		status = nil
+	end
+	
+	return success ~= nil, status, respbody, headers
+end
+
+
+
+
+
+return {
+	["<static> cUrlClient:Delete(string, function)"] = function(a_Simulator, a_Self, a_Url, a_Callback)
+		local success, status, body, headers = MakeRequest("DELETE", a_Url)
+		a_Simulator:processCallbackRequest({
```

Comment:
With #18 merged, you should now use `queueCallbackRequest` instead of `processCallbackRequest` - that makes the callback queued and executed after this call returns, which catches more plugin errors.

Sorry for the incrementality of the changes I request, I had no idea #18 would be implemented so fast.